ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -765,7 +765,7 @@ generateGhcLibCabal ghcFlavor = do
         , "bug-reports: https://github.com/digital-asset/ghc-lib/issues"
         , "data-dir: " ++ dataDir
         , "data-files:"] ++ indent dataFiles ++
-        [ "extra-source-files:"] ++ indent (ghcLibExtraFiles ghcFlavor) ++
+        [ "extra-source-files:"] ++ indent (ghcLibExtraFiles ghcFlavor \\ [hadrianGeneratedRoot ghcFlavor </> "ghcversion.h"]) ++
```

Comment:
why do we need to exclude this file? `ghcversion.h` has existed for a long time hasn’t it? did it change place?

Reply:
no, we've just been fortunate before now. what i determined is, there is a CMM .x file (lexer file). it now contains logic conditional on __GLASGOW_HASKELL__. if you build ghc-lib with a non ghc-9.0 via cabal (not stack), it's clear that in this configuration ghcversion.h gets in the way of the compiler provided ghc version.  so in summary, we need ghcversion.h to produce the dist but when building the dist, when you think about it, there is no scenario when substituting for __GLASGOW_HASKELL__ is the right thing to do - ghcversion.h should not be shipped.

Reply:
btw, in case you were wondering, ghcversion.h literally only contains a definition of `__GLASGOW_HASKELL__` and related constants. nothing else.

Reply:
makes sense, thanks for the explanation!